### PR TITLE
Avoid `new $this->modelClass` in ActiveQuery::findWith()

### PR DESCRIPTION
**Pull request by cebe**
reduces the issue described in #8639 to `joinWith()` and `asArray()`. Everything else should work then.

### CODE REVIEW

**Comment by cebe** [comment]
similar to #10939

**Comment by SilverFire** [comment]
:+1:

File: framework/db/ActiveQueryTrait.php [hunk 1]
```diff
@@ -159,7 +159,10 @@ private function createModels($rows)
      */
     public function findWith($with, &$models)
     {
-        $primaryModel = new $this->modelClass;
+        $primaryModel = reset($models);
+        if (!$primaryModel instanceof ActiveRecordInterface) {
+            $primaryModel = new $this->modelClass;
```
Comment (SamMousa) [inline framework/db/ActiveQueryTrait.php]: Why not use `Yii::createObject` here?

Comment (cebe) [inline framework/db/ActiveQueryTrait.php]: we do not use it in any place inside AR currently so it would need to be changed in many places. I am thinking about moving the responsibility of creating relation objects to the AR class to fix #8639 but that is out of scope of this PR.

Comment (SamMousa) [inline framework/db/ActiveQueryTrait.php]: Ah, so avoiding is referencing the fact that in lots of cases basically you remove the need to instantiate a new object, not the fact that you use `new` when you have to instantiate an object.

Comment (cebe) [inline framework/db/ActiveQueryTrait.php]: exactly. this already helps a lot as you can now use everything except joinWith and asArray. Will address the other issue in another PR.

**Comment by cebe** [comment]
merged.
